client/src/views/Home.js

```diff
@@ -1,50 +1,74 @@
-import React, { Component } from "react";
+import React, {Component} from "react";
 import "../css/Home.css";
 
 export default class Home extends Component {
 
-    constructor(props) {
-        super(props);
-        this.state = {value: ''};
+  constructor(props) {
+    super(props);
+    this.state = {email: '', error: ''};
 
-        this.handleChange = this.handleChange.bind(this);
-        this.handleSubmit = this.handleSubmit.bind(this);
-    }
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
 
-    handleChange(event) {
-        this.setState({value: event.target.value});
-    }
+  handleChange(event) {
+    this.setState({email: event.target.value});
+  }
+
+  async handleSubmit(event) {
+    const response = await fetch('http://localhost:5000/mailing_list?email=' + this.state.email, {
+      method: 'POST',
+      headers: {
+        Accept: 'application/json',
+        'Content-Type': 'application/json',
+      }
+    });
 
-    handleSubmit(event) {
-        alert('An email was submitted: ' + this.state.value);
-        event.preventDefault();
+    const body = await response.json();
+
+    if (response.status !== 201) {
+      if (body.error) {
+        this.setState({error: body.error});
+      }
+      else {
+        this.setState({error: JSON.stringify(body)});
+      }
+    } else {
+      this.setState({error: ''});
+      alert('An email was submitted: ' + this.state.email);
     }
+    event.preventDefault();
```

Comment:
is the preventDefault() necessary? I get this error in the console whenever I click the button: 
<img width="1318" alt="screen shot 2019-01-04 at 9 38 57 am" src="https://user-images.githubusercontent.com/12488887/50702252-34e7d400-1005-11e9-9112-3b910fef385c.png">